server/datastore/mysql/software.go

```diff
@@ -3139,8 +3139,10 @@ func (ds *Datastore) SetHostSoftwareInstallResult(ctx context.Context, result *f
 			return ctxerr.Wrap(ctx, notFound("HostSoftwareInstall").WithName(result.InstallUUID), "host software installation not found")
 		}
 
-		if _, err := ds.activateNextUpcomingActivity(ctx, tx, result.HostID, result.InstallUUID); err != nil {
-			return ctxerr.Wrap(ctx, err, "activate next activity")
+		if result.Status() != fleet.SoftwareInstallPending {
```

Comment:
@mna, from the Slack discussion, I understand that when orbit fails to download the installer, it tries to save an empty response. And the fix here is that will get treated as a pending status so it stays at the top of the pending activities and orbit will try the download again. Do I understand that correctly? 

Does that mean if orbit gets stuck in a loop where it repeatedly fails to download the installer, this will be stuck in pending and block all upcoming activities?

Reply:
Yes, that's correct, I do make a mention of this in the video - that's also how it behaves pre-unified queue. I think ultimately the fix would be the cancellation option of upcoming activities, or a new feature that automatically marks as failed if it can't download after a number of attempts.